File: Data Structures & Algorithms/subsets-ii/submission-0.java
class Solution {
    private List<List<Integer>> ans;
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        this.ans = new ArrayList<>();
        helper(nums,new ArrayList<>(), 0);
        return ans;
    }
    private void helper( int[] nums,List<Integer> curr, int index)
    {
        ans.add(new ArrayList<>(curr));
        for(int i=index; i<nums.length; i++)
        {
            if (i > index && nums[i] == nums[i-1])
                continue;   
            curr.add(nums[i]);
            helper(nums,curr, i+1);
            curr.remove(curr.size() - 1);    
        }
    }
}

File: Data Structures & Algorithms/palindrome-partitioning/submission-0.py
class Solution:
    def partition(self, s: str) -> List[List[str]]:
        res=[]
        def helper(curr, index):
            if index == len(s):
                res.append(curr[:])
                return
            for i in range(index, len(s)):
                temp=s[index : i+1]
                if temp == temp[:: -1]:
                    curr.append(temp)
                    helper(curr, i+1)
                    curr.pop()
        helper([], 0)
        return res            